checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -349,6 +349,13 @@ public Void visitNewArray(NewArrayTree tree, Void p) {
                     type.toString());
         }
 
+        List<? extends AnnotationMirror> annotations =
+                TreeUtils.annotationsFromArrayCreation(tree, 0);
+        if (AnnotationUtils.containsSame(annotations, NULLABLE)
+                || AnnotationUtils.containsSame(annotations, MONOTONIC_NONNULL)
+                || AnnotationUtils.containsSame(annotations, POLYNULL)) {
+            checker.reportWarning(tree, "new.array.nullable.ignored");
```

Comment:
```suggestion
            checker.reportError(tree, "new.array.nullable.ignored");
```

Related to the discussion of the error key, the other `nullness.on` messages are errors.